crates/reth-rbuilder/src/main.rs

```diff
@@ -0,0 +1,129 @@
+//! `rbuilder` running in-process with vanilla reth.
+//!
+//! Usage: `cargo run -r --bin reth-rbuilder -- node --rbuilder.config <path-to-your-config-toml>`
+
+use clap::Args;
+use rbuilder::{
+    live_builder::{base_config::load_config_toml_and_env, cli::LiveBuilderConfig, config::Config},
+    telemetry,
+};
+use reth_db_api::Database;
+use reth_provider::{
+    providers::BlockchainProvider, DatabaseProviderFactory, HeaderProvider, StateProviderFactory,
+};
+use std::{path::PathBuf, process};
+use tokio::task;
+use tracing::error;
+
+// Prefer jemalloc for performance reasons.
+#[cfg(all(feature = "jemalloc", unix))]
+#[global_allocator]
+static ALLOC: tikv_jemallocator::Jemalloc = tikv_jemallocator::Jemalloc;
+
+#[derive(Debug, Clone, Args, PartialEq, Eq, Default)]
+pub struct ExtraArgs {
+    /// Path of the rbuilder config to use
+    #[arg(long = "rbuilder.config")]
+    pub rbuilder_config: PathBuf,
+    /// Enable the engine2 experimental features on reth binary
+    #[arg(long = "engine.experimental", default_value = "false")]
+    pub experimental: bool,
+}
+
+fn main() {
+    use clap::Parser;
+    use reth::cli::Cli;
+    use reth_node_builder::EngineNodeLauncher;
+    use reth_node_ethereum::{node::EthereumAddOns, EthereumNode};
+    use reth_provider::providers::BlockchainProvider2;
+
+    reth_cli_util::sigsegv_handler::install();
+
+    if let Err(err) = Cli::<ExtraArgs>::parse().run(|builder, extra_args| async move {
+        let enable_engine2 = extra_args.experimental;
+        match enable_engine2 {
```

Comment:
Is there any reason to use the legacy engine when running as a Reth module (out of being an experimental feature)?

Reply:
Good q, I'm not familiar with it enough to known. Figured I'd leave it as an option since that is the default in reth 1.0.6.